### PR TITLE
Addition of shit-list.lic

### PR DESCRIPTION
**Pull request by jayRyan24**
```
>;shit help
--- Lich: shit-list active.
 
  ;shit-list add name desc
   add          Add name to list. 
   name         Name to add.
   desc         Short description of why they were added in quotes.
  ;shit-list find name
   find         Find if name exists and gives their description.
   name         Name of character to find.
 
  ;shit-list rem name
   rem          Remove name from list.
   name         Name of character to remove. 
 
  ;shit-list clear confirm
   clear        Clear list! WARNING! 
   confirm      CONFIRM YOUR CHOICE!.
 
  ;shit-list
--- Lich: shit-list has exited.
```

```
>;shit
--- Lich:
 shit-list active.
Current Shitters:
--- Lich: shit-list has exited.
```

```
>;shit add chuno "because he ain't me"
--- Lich: shit-list active.
Added chuno, because he ain't me to your shit list!
--- Lich: shit-list has exited.
```

```
>;shit
--- Lich: shit-list active.
Current Shitters:
Name: Chuno         Desc: Because he ain't me
--- Lich: shit-list has exited.
```

```
;shit
--- Lich: shit-list active.
Current Shitters:
Name: Aardvark      Desc: To test sorting alphabetically
Name: Ardvark       Desc: Same as aardvark
Name: Chuno         Desc: Because he ain't me
Name: Leustyin      Desc: Because he doesn't love me like i love him
Name: Natarian      Desc: Because this shitter lit me on fire
--- Lich: shit-list has exited.
```

```
>;shit find leu
--- Lich: shit-list active.
Found the shitter(s)!
Name: Leustyiin     Desc: Testing the second case
Name: Leustyin      Desc: Because i don't like him
--- Lich: shit-list has exited.
```

```
--- Lich: shit-list active.
>;shit find nat
>
Found the shitter(s)!
Name: Natarian    Desc: Because this shitter lit me on fire
--- Lich: shit-list has exited.
```

```
;shit rem leustyin
--- Lich: shit-list active.
Removed leustyin from your Shit List!
--- Lich: shit-list has exited.
```

```
>;shit
--- Lich: shit-list active.
Current Shitters:
Name: Aardvark      Desc: To test sorting alphabetically
Name: Ardvark       Desc: Same as aardvark
Name: Chuno         Desc: Because he ain't me
Name: Natarian      Desc: Because this shitter lit me on fire
--- Lich: shit-list has exited.
```

```
>profile leustyin
             Name: Leustyin (SL)
           Spouse: No Info
  Roleplay Stance: None
       PvP Stance: Open
```

### CODE REVIEW

**Comment by jayRyan24** [comment]
I could just make this a player-list, and you can just keep handy notes about people, etc, as well.

File: shit-list.lic [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    @file_name = "#{checkname}_shit.list"
+
+    if args.add
+      add_new_shitter(args.name.downcase, args.desc)
+    elsif args.find
+      find_shitter_in_list(args.name.downcase)
+    elsif args.rem
+      remove_shitter(args.name.downcase)
+    elsif args.clear
+      clear_list(args.confirm)
```
Comment (rpherbig) [inline shit-list.lic]: 💡 Why pass in `args.confirm`? The arg parsing should ensure it's always there (since it is not marked as optional).

Comment (jayRyan24) [reply shit-list.lic]: I know, it was something I did early and just didn't remove, lol.

File: shit-list.lic [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      pause 2
+      start_script('textsubs')
+    end
+  end
+
+  def find_shitter_in_list(name_in)
+    output = [] << "Found the shitter(s)!"
+
+    File.open(@file_name, "r") do |file|
+      file.each { |line|
```
Comment (rpherbig) [inline shit-list.lic]: 💡 You could pull the `if` out into a method chain, something like:

```ruby
file.select { |line| line.split(',').first.include?(name_in) }
      .each { |line|
          name = line.split(',').first
          line.slice!(name + ', ')
          output << format("Name: %s    Desc: %s", name.capitalize.ljust(10), line.capitalize)
      }
}
```

Comment (jayRyan24) [reply shit-list.lic]: Ah, yes, probably would have been cleaner. Thanks.

**Pull request merged**